core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -974,8 +973,39 @@ private[kafka] class Processor(val id: Int,
                 val context = new RequestContext(header, connectionId, channel.socketAddress,
                   channel.principal, listenerName, securityProtocol,
                   channel.channelMetadataRegistry.clientInformation, isPrivilegedListener)
-                val req = new RequestChannel.Request(processor = id, context = context,
-                  startTimeNanos = nowNanos, memoryPool, receive.payload, requestChannel.metrics)
+
+                val principalSerde = Option(channel.principalSerde.orElse(null))
```

Comment:
nit: you can just use `channel.principalSerde.asScala`

Reply:
Had a try but it seems java Optional doesn't have an `asScala` option

Reply:
You probably need the following:
```scala
import scala.compat.java8.OptionConverters._
```